### PR TITLE
Reader: Fix various embeds

### PR DESCRIPTION
**Pull request by blowery**
Fixes some of the busted embeds listed found in internal testing. The remaining busted embeds require special script handling to work.

### CODE REVIEW

**Comment by fraying** [comment]
Nice! Testing https://calypso.live/read/feeds/49566081 ...

Fixed: Wistia, Twitch, Internet Archive Video, Internet Archive Ebook, CodePen
Still To Do: Getty, Issuu, Google+, Pinterest

File: client/blocks/reader-related-card-v2/index.jsx [hunk 1]
```diff
@@ -26,17 +26,18 @@ const RELATED_IMAGE_WIDTH = 385; // usual width of featured images in related po
 function AuthorAndSiteFollow( { post, site, onSiteClick, followSource } ) {
 	const siteUrl = getStreamUrl( post.feed_ID, post.site_ID );
 	const siteName = ( site && site.title ) || post.site_name;
-	const authorAndSiteAreDifferent = ! areEqualIgnoringWhitespaceAndCase( siteName, post.author.name );
+	const authorName = get( post, 'author.name', '' );
```
Comment (blowery) [inline client/blocks/reader-related-card-v2/index.jsx]: i included this fix here because it's required to test the IA video from the test stream.

**Comment by samouri** [comment]
Code LGTM. Testing now.

File: client/blocks/reader-related-card-v2/index.jsx (hunk 1 above)
Comment (samouri) [inline client/blocks/reader-related-card-v2/index.jsx]: What is this fixing? Is there an issue with the way `areEqualIgnoringWhitespaceAndCase` functions?

Comment (blowery) [inline client/blocks/reader-related-card-v2/index.jsx]: mostly fixes usage of author.name later on without checking that `author` is truthy. `areEqualIgnoringWhitespaceAndCase` also has an issue with taking undefined arguments.

**Comment by samouri** [comment]
Confirmed same ones fixed as described by @fraying.

I'd like to add one more to the list: https://github.com/Automattic/wp-calypso/issues/9660
Github Gists.

**Pull request merged**